pkg/cloudprovider/aws.go

```diff
@@ -45,7 +46,7 @@ func (a *AWS) initCredentials() error {
 	}
 	if a.cfg.APIOverride != "" {
 		customResolver := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-			if service == endpoints.Ec2ServiceID {
```

Comment:
to be more specific, a pointer to:

https://github.com/aws/aws-sdk-go/blob/17200eb6399cd12e0baa44e0973398ee0432ab17/aws/endpoints/dep_service_ids.go#L5-L6

```
// Service identifiers
//
// Deprecated: Use client package's EndpointsID value instead of these
// ServiceIDs. These IDs are not maintained, and are out of date.
```

Reply:
thanks for the good review. I've updated the commit message to be more clear.